BookingsApi/BookingsApi/Controllers/JudiciaryPersonController.cs

```diff
@@ -0,0 +1,88 @@
+        private readonly IQueryHandler _queryHandler;
+        private readonly ICommandHandler _commandHandler;
+        private readonly ILogger<JudiciaryPersonController> _logger;
+        
+        public JudiciaryPersonController(IQueryHandler queryHandler, ICommandHandler commandHandler, ILogger<JudiciaryPersonController> logger)
+        {
+            _queryHandler = queryHandler;
+            _commandHandler = commandHandler;
+            _logger = logger;
+        }
+        
+        [HttpPost]
+        [OpenApiOperation("BulkJudiciaryPersons")]
+        [ProducesResponseType((int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        [ProducesResponseType(typeof(BulkJudiciaryPersonResponse), (int)HttpStatusCode.OK)]
+        public async Task<IActionResult> BulkJudiciaryPersonsAsync(IEnumerable<JudiciaryPersonRequest> request)
+        {
+            var bulkResponse = new BulkJudiciaryPersonResponse();
+            
+            foreach (var item in request)
+            {
+                var validation = await new JudiciaryPersonRequestValidation().ValidateAsync(item);
+                if (!validation.IsValid)
+                {
+                    bulkResponse.ErroredRequests.Add(new JudiciaryPersonErrorResponse
+                    {
+                        Message = $"{string.Format(BulkItemErrorMessage, item.Id)} - {string.Join(", ", validation.Errors.Select(x => x.ErrorMessage))}", 
+                        JudiciaryPersonRequest = item
+                    });
+                    
+                    continue;
+                }
+            
+                try
+                {
+                    var query = new GetJudiciaryPersonByExternalRefIdQuery(item.Id);
+                    var judiciaryPerson = await _queryHandler.Handle<GetJudiciaryPersonByExternalRefIdQuery, JudiciaryPerson>(query);
+
+                    if (judiciaryPerson == null)
+                    {
+                        await _commandHandler.Handle(new AddJudiciaryPersonCommand(item.Id, item.PersonalCode, item.Title, item.KnownAs, item.Surname,
+                            item.Fullname, item.PostNominals, item.Email));
+                    }
+                    else
+                    {
+                        await _commandHandler.Handle(new UpdateJudiciaryPersonByExternalRefIdCommand(item.Id, item.PersonalCode, 
+                            item.Title, item.KnownAs, item.Surname, item.Fullname, item.PostNominals, item.Email));
+                    }
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, string.Format(BulkItemErrorMessage, item.Id));
+                    bulkResponse.ErroredRequests.Add(new JudiciaryPersonErrorResponse
+                    {
+                        Message = BulkItemErrorMessage, JudiciaryPersonRequest = item
+                    });
+                }
+            }
```

Comment:
could this be moved to a service? i know the existing controllers have the stuff dumped in there but this is a new controller, we could try and start well?

Reply:
I'm not sure why this needs to be in a new service, its the controller actions responsibility to build up its response from the result of the service call. I think we should keep it simple... until someone needs to extend the response functionality

Reply:
I am leaning with @levibutler. The controller would look a lot leaner and cleaner by having a service responsible for adding or updating. The controller should validate the request and then pass on the work.

Reply:
Sorry, I don't agree, I'm adding a response object to a list and passing that back in the response.
Do you want the overhead of code running to a new IoC'ed service to .Add() item to a list.
There is one method in the controller, if there was 2 and common functionality I would tend to agree.

Reply:
ok lets leave for now